tests.js

```diff
@@ -71,10 +90,22 @@ QUnit.module("Logo Unit Tests", {
 
     this.assert_stream = function(expression, expected) {
       this.stream.clear();
-      this.interpreter.run(expression, {returnResult: true});
-      var actual = this.stream.outputbuffer;
-      this.stream.clear();
-      t.equal(actual, expected, expression);
+      var result = this.interpreter.run(expression, {returnResult: true});
+      if (! (result && result.then)) {
```

Comment:
Unnecessary - you can always just write `result = Promise.resolve(result)` which will be a no-op if already a Promise.